File: Medium/Check if frequencies can be equal/check-if-frequencies-can-be-equal.py
#User function Template for python3
class Solution:
    def sameFreq(self, s):
        mp, freq = {}, {}
        
        for c in s:
            if c in mp:
                mp[c] += 1
            else:
                mp[c] = 1
        
        max_freq, min_freq = float('-inf'), float('inf')
        for key, value in mp.items():
            if value in freq:
                freq[value] += 1
            else:
                freq[value] = 1
            max_freq = max(max_freq, value)
            min_freq = min(min_freq, value)
            
        if len(freq) != 1 and freq[max_freq] != 1:
            return False
        if max_freq - min_freq > 1:
            return False
        
        return True


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == "__main__":
	T=int(input())

	for _ in range(T):
		s = input()
		ob = Solution()
		answer = ob.sameFreq(s)
		if answer:
			print(1)
		else:
			print(0)

# } Driver Code Ends

File: Medium/Check if frequencies can be equal/check-if-frequencies-can-be-equal.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	bool sameFreq(string s) {
	    // code here 
	    unordered_map<char, int> mp;
	    unordered_map<int, int> freq;
	    for(char c : s) {
	        mp[c]++;
	    }
	    
	    int min_freq = 1e8, max_freq = -1;
	    for(auto it : mp) {
	        freq[it.second]++;
	        max_freq = max(max_freq, it.second);
	        min_freq = min(min_freq, it.second);
	    }
	    
	    if(max_freq - min_freq > 1) return false;
	    if(freq.size() != 1 && freq[max_freq] != 1) return false;
	    return true;
	}
};

//{ Driver Code Starts.
int main(){
    int t;
    cin>>t;
    while(t--)
    {
        string s;
        cin>>s;
        Solution ob;
        cout<<ob.sameFreq(s)<<endl;
    }
}



// } Driver Code Ends